src/crimson/os/seastore/extent_placement_manager.h

```diff
@@ -545,6 +545,15 @@ class ExtentPlacementManager {
     return background_process.run_until_halt();
   }
 
+  bool get_checksum_needed(paddr_t addr) {
+    // checksum offloading only for blocks physically stored in the device
+    assert(addr.is_absolute() || addr.is_fake() || addr.is_relative() || addr.is_root());
```

Comment:
See https://github.com/ceph/ceph/pull/57782#discussion_r1677268711

Let's move `addr.is_relative()` out and handle it correctly instead.

Also move `addr.is_root()` out to the user in `Cache::complete_commit()` (I suppose this is the only possible case).